src/CabanaPD_Solver.hpp

```diff
@@ -254,6 +244,33 @@ class SolverNoFracture
             particles->output( 0, 0.0, output_reference );
     }
 
+    template <std::size_t NumPrenotch>
+    void init_prenotch( Prenotch<NumPrenotch> prenotch )
+    {
+        // Create prenotch.
+        auto mu = force->getBrokenBonds();
+        prenotch.create( exec_space{}, mu, *particles, force->_neigh_list );
```

Comment:
For your consideration, in lieu of SFINAE init_prenotch, why not pass the prenotch to the force model, not the other way around?